libs/configuration/src/lib/configuration/configuration.component.ts

```diff
@@ -1,11 +1,75 @@
 import { CommonModule } from '@angular/common';
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
+import { Project } from '@angular-cli-gui/shared/data';
+import { JsonObject } from '@angular-devkit/core/src/json/utils';
+import { FormlyFieldConfig, FormlyModule } from '@ngx-formly/core';
+import { FormlyJsonschema } from '@ngx-formly/core/json-schema';
+import { FormlyMaterialModule } from '@ngx-formly/material';
+import { combineLatest, map, Subject, switchMap, tap } from 'rxjs';
+
+import { WorkspaceSettingsService } from '../services';
 
 @Component({
   selector: 'cli-configuration',
   standalone: true,
-  imports: [CommonModule],
+  imports: [
+    CommonModule,
+    FormlyModule,
+    FormlyMaterialModule,
+    ReactiveFormsModule,
+  ],
   templateUrl: './configuration.component.html',
   styleUrls: ['./configuration.component.css'],
 })
-export class ConfigurationComponent {}
+export class ConfigurationComponent implements OnInit {
+  private formFields$ = new Subject<FormlyFieldConfig[]>();
+  private formData$ = new Subject<JsonObject>();
+  private projectName = '';
+
+  formly$ = combineLatest({
+    data: this.formData$,
+    field: this.formFields$,
+  });
+
+  form = this.fb.group({});
+
+  constructor(
+    private workspaceSettingsService: WorkspaceSettingsService,
+    private fb: FormBuilder,
+    private schema: FormlyJsonschema
+  ) {}
+
+  ngOnInit(): void {
+    this.workspaceSettingsService
+      .readWorkspaceProjectConfiguration()
+      .subscribe((configuration) => {
+        const filedGroup = this.schema.toFieldConfig(configuration)
+          .fieldGroup as FormlyFieldConfig[];
+        this.formFields$.next(filedGroup);
+      });
+    this.workspaceSettingsService
+      .readWorkspaceProjectNames()
+      .pipe(
+        map((names) => names[0]),
+        tap((projectName) => (this.projectName = projectName)),
+        switchMap((currentProjectName) =>
+          this.workspaceSettingsService.readWorkspaceProject(
+            currentProjectName as string
+          )
+        )
+      )
+      .subscribe((formData) => {
+        this.formData$.next(formData);
+      });
```

Comment:
Maybe this can be cleaned up a bit, and also reduce the risk of future race conditions and also group up value emitions
```suggestion
readonly projectConfiguration$ =
    this.workspaceSettingsService.readWorkspaceProjectConfiguration();

  readonly projectNames$ =
    this.workspaceSettingsService.readWorkspaceProjectNames();

  readonly workspaceProject$ = this.projectNames$.pipe(
    map((names) => names[0]),
    tap((projectName) => (this.projectName = projectName)),
    switchMap((currentProjectName: string) =>
      this.workspaceSettingsService.readWorkspaceProject(currentProjectName)
    )
  );
  constructor(
    private workspaceSettingsService: WorkspaceSettingsService,
    private fb: FormBuilder,
    private schema: FormlyJsonschema
  ) {}
  ngOnInit(): void {
    forkJoin([projectConfiguration$, project$]).subscribe(
      ([configuration, project]) => {
        const fields = this.schema.toFieldConfig(configuration)
          .fieldGroup as FormlyFieldConfig[];
        this.formFields$.next(fields);
        this.formData$.next(project);
      }
    );
  }
```

Reply:
Probably worth adding error handling too

Reply:
Consider merging fields and data to the same subject if they are updated at the same time to reduce change detection cycles

Reply:
like the approached.. fixed! 
though in the future we will have state and i will change the code little bit